corehq/apps/accounting/models.py

```diff
@@ -1230,6 +1232,13 @@ def send_ending_reminder_email(self):
                     'email': email,
                 })
 
+    def set_billing_account_entry_point(self):
+        no_current_entry_point = self.account.entry_point == EntryPoint.NOT_SET
+        self_serve = self.service_type == SubscriptionType.SELF_SERVICE
+        if (no_current_entry_point and self_serve and not self.is_trial):
+            self.account.entry_point = EntryPoint.SELF_STARTED
+            self.account.save()
```

Comment:
Heh, that's the challenge - ensuring the account gets saved is at odds with ensuring that it doesn't accidentally or unnecessarily get saved. I think either implementation is fine, it's a matter of what the developer is going to expect to happen, which depends on what's typical in HQ.

Which is a total non-answer, but I haven't read enough of the HQ codebase to know where we usually do saving.